<ui version="4.0" >
 <class>mainWindow</class>
 <widget class="QMainWindow" name="mainWindow" >
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>896</width>
    <height>738</height>
   </rect>
  </property>
  <property name="sizePolicy" >
   <sizepolicy vsizetype="Preferred" hsizetype="Expanding" >
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle" >
   <string>OpenSolid</string>
  </property>
  <property name="windowIcon" >
   <iconset/>
  </property>
  <widget class="QWidget" name="centralWidget" >
   <property name="layoutDirection" >
    <enum>Qt::LeftToRight</enum>
   </property>
   <layout class="QHBoxLayout" >
    <property name="spacing" >
     <number>6</number>
    </property>
    <property name="leftMargin" >
     <number>6</number>
    </property>
    <property name="topMargin" >
     <number>6</number>
    </property>
    <property name="rightMargin" >
     <number>6</number>
    </property>
    <property name="bottomMargin" >
     <number>6</number>
    </property>
    <item>
     <widget class="QFrame" name="modelViewFrame" >
      <property name="sizePolicy" >
       <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="frameShape" >
       <enum>QFrame::StyledPanel</enum>
      </property>
      <property name="frameShadow" >
       <enum>QFrame::Plain</enum>
      </property>
      <layout class="QVBoxLayout" >
       <property name="spacing" >
        <number>6</number>
       </property>
       <property name="leftMargin" >
        <number>1</number>
       </property>
       <property name="topMargin" >
        <number>1</number>
       </property>
       <property name="rightMargin" >
        <number>1</number>
       </property>
       <property name="bottomMargin" >
        <number>1</number>
       </property>
       <item>
        <widget class="modelView" native="1" name="currentModelView" >
         <property name="sizePolicy" >
          <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar" >
   <property name="geometry" >
    <rect>
     <x>0</x>
     <y>0</y>
     <width>896</width>
     <height>34</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuView" >
    <property name="title" >
     <string>View</string>
    </property>
    <addaction name="actionOrtho" />
   </widget>
   <widget class="QMenu" name="menuFile" >
    <property name="title" >
     <string>File</string>
    </property>
    <addaction name="actionNewProject" />
    <addaction name="actionOpenProject" />
    <addaction name="separator" />
    <addaction name="actionSaveProject" />
    <addaction name="actionSaveProjectAs" />
    <addaction name="separator" />
    <addaction name="actionQuit" />
   </widget>
   <addaction name="menuFile" />
   <addaction name="menuView" />
  </widget>
  <widget class="QStatusBar" name="statusbar" >
   <property name="autoFillBackground" >
    <bool>false</bool>
   </property>
  </widget>
  <widget class="QToolBar" name="mainToolBar" >
   <property name="windowTitle" >
    <string>Main Tool Bar</string>
   </property>
   <property name="movable" >
    <bool>true</bool>
   </property>
   <property name="orientation" >
    <enum>Qt::Horizontal</enum>
   </property>
   <attribute name="toolBarArea" >
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <addaction name="separator" />
   <addaction name="actionSaveProject" />
  </widget>
  <widget class="QDockWidget" name="dockProjectControl" >
   <property name="floating" >
    <bool>false</bool>
   </property>
   <property name="features" >
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable|QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="allowedAreas" >
    <set>Qt::LeftDockWidgetArea|Qt::NoDockWidgetArea|Qt::RightDockWidgetArea</set>
   </property>
   <property name="windowTitle" >
    <string>Project</string>
   </property>
   <attribute name="dockWidgetArea" >
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockProjectContents" >
    <layout class="QVBoxLayout" >
     <property name="leftMargin" >
      <number>6</number>
     </property>
     <property name="topMargin" >
      <number>3</number>
     </property>
     <property name="rightMargin" >
      <number>6</number>
     </property>
     <property name="bottomMargin" >
      <number>6</number>
     </property>
     <item>
      <widget class="QTreeView" name="graphTreeView" >
       <property name="editTriggers" >
        <set>QAbstractItemView::AllEditTriggers</set>
       </property>
       <property name="dragEnabled" >
        <bool>true</bool>
       </property>
       <property name="dragDropMode" >
        <enum>QAbstractItemView::InternalMove</enum>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockProperties" >
   <property name="minimumSize" >
    <size>
     <width>0</width>
     <height>200</height>
    </size>
   </property>
   <property name="features" >
    <set>QDockWidget::DockWidgetFloatable|QDockWidget::DockWidgetMovable|QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <property name="windowTitle" >
    <string>Properties</string>
   </property>
   <attribute name="dockWidgetArea" >
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2" >
    <layout class="QVBoxLayout" >
     <property name="leftMargin" >
      <number>6</number>
     </property>
     <property name="topMargin" >
      <number>3</number>
     </property>
     <property name="rightMargin" >
      <number>6</number>
     </property>
     <property name="bottomMargin" >
      <number>6</number>
     </property>
     <item>
      <widget class="QTableView" name="tblProperties" >
       <property name="acceptDrops" >
        <bool>true</bool>
       </property>
       <property name="editTriggers" >
        <set>QAbstractItemView::AllEditTriggers</set>
       </property>
       <property name="selectionMode" >
        <enum>QAbstractItemView::SingleSelection</enum>
       </property>
       <property name="wordWrap" >
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QToolBar" name="viewToolBar" >
   <property name="windowTitle" >
    <string>View Tool Bar</string>
   </property>
   <property name="orientation" >
    <enum>Qt::Vertical</enum>
   </property>
   <property name="iconSize" >
    <size>
     <width>22</width>
     <height>22</height>
    </size>
   </property>
   <attribute name="toolBarArea" >
    <enum>RightToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <addaction name="actionOrtho" />
   <addaction name="actionAlignXY" />
   <addaction name="actionAlignYZ" />
   <addaction name="actionAlignXZ" />
   <addaction name="actionAlignCustom" />
  </widget>
  <widget class="QToolBar" name="graphToolBar" >
   <property name="windowTitle" >
    <string>Graph Tool Bar</string>
   </property>
   <property name="iconSize" >
    <size>
     <width>22</width>
     <height>22</height>
    </size>
   </property>
   <attribute name="toolBarArea" >
    <enum>LeftToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak" >
    <bool>false</bool>
   </attribute>
   <addaction name="actionAddComponent" />
   <addaction name="actionAddFragment" />
   <addaction name="actionAddPoint" />
   <addaction name="actionAddCurve" />
   <addaction name="actionAddSurface" />
   <addaction name="actionAddParameter" />
  </widget>
  <action name="actionQuit" >
   <property name="text" >
    <string>Quit</string>
   </property>
  </action>
  <action name="actionOrtho" >
   <property name="checkable" >
    <bool>true</bool>
   </property>
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/ortho.png</iconset>
   </property>
   <property name="text" >
    <string>Orthographic View</string>
   </property>
   <property name="toolTip" >
    <string>Toggle Orthographic View</string>
   </property>
  </action>
  <action name="actionOpenProject" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/document-open.png</iconset>
   </property>
   <property name="text" >
    <string>Open Project</string>
   </property>
  </action>
  <action name="actionNewProject" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/document-new.png</iconset>
   </property>
   <property name="text" >
    <string>New Project</string>
   </property>
  </action>
  <action name="actionSaveProject" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/document-save.png</iconset>
   </property>
   <property name="text" >
    <string>Save Project</string>
   </property>
  </action>
  <action name="actionSaveProjectAs" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/document-save-as.png</iconset>
   </property>
   <property name="text" >
    <string>Save Project As</string>
   </property>
  </action>
  <action name="actionAlignXY" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/align_xy.png</iconset>
   </property>
   <property name="text" >
    <string>Align To X - Y Axis</string>
   </property>
   <property name="iconText" >
    <string>Align X - Y</string>
   </property>
  </action>
  <action name="actionAlignXZ" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/align_xz.png</iconset>
   </property>
   <property name="text" >
    <string>Align To X - Z Axis</string>
   </property>
  </action>
  <action name="actionAlignYZ" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/align_yz.png</iconset>
   </property>
   <property name="text" >
    <string>Align To Y - Z Axis</string>
   </property>
  </action>
  <action name="actionAlignCustom" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/align_custom.png</iconset>
   </property>
   <property name="text" >
    <string>Custom To Align Axis</string>
   </property>
  </action>
  <action name="actionAddComponent" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/component_add.png</iconset>
   </property>
   <property name="text" >
    <string>Add Component</string>
   </property>
  </action>
  <action name="actionAddFragment" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/fragment_add.png</iconset>
   </property>
   <property name="text" >
    <string>Add Fragment</string>
   </property>
  </action>
  <action name="actionAddPoint" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/point_add.png</iconset>
   </property>
   <property name="text" >
    <string>Add Point</string>
   </property>
  </action>
  <action name="actionAddCurve" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/curve_add.png</iconset>
   </property>
   <property name="text" >
    <string>Add Curve</string>
   </property>
  </action>
  <action name="actionAddSurface" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/surface_add.png</iconset>
   </property>
   <property name="text" >
    <string>Add Surface</string>
   </property>
  </action>
  <action name="actionAddParameter" >
   <property name="icon" >
    <iconset resource="res/images.qrc" >:/images/param_new.png</iconset>
   </property>
   <property name="text" >
    <string>Add Parameter</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>modelView</class>
   <extends>QWidget</extends>
   <header>modelView.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="res/images.qrc" />
 </resources>
 <connections/>
</ui>
